auth/providers/azure/pop_tokenverifier.go

```diff
@@ -27,23 +27,66 @@ import (
 	"math/big"
 	"strconv"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/pkg/errors"
 	"gopkg.in/square/go-jose.v2/jwt"
 	"k8s.io/klog/v2"
 )
 
+// create a cache to save nonce claim to make sure the nonce is not reused
+var nonceMap = nonceCache{v: make(map[string]time.Time)}
+
+type nonceCache struct {
+	mu sync.Mutex
+	v  map[string]time.Time
+	// counter to keep track of the number of nonce claims in the cache
+	counter int
+}
+
+func (c *nonceCache) Dec(key string) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.counter--
+}
```

Comment:
This function isn't used (and makes the counter inconsistent).

Reply:
yep I did clean it